src/Libraries/RevitNodes/Elements/TextElement.cs

```diff
@@ -0,0 +1,239 @@
+        }
+
+        /// <summary>
+        /// Set Internal Element
+        /// </summary>
+        /// <param name="element"></param>
+        internal void InternalSetElement(Autodesk.Revit.DB.Element element)
+        {
+            InternalRevitElement = element;
+            InternalElementId = element.Id;
+            InternalUniqueId = element.UniqueId;
+        }
+
+        /// <summary>
+        /// Set Internal Text Display Settings
+        /// </summary>
+        /// <param name="value"></param>
+        /// <param name="bold"></param>
+        /// <param name="italic"></param>
+        /// <param name="size"></param>
+        /// <param name="fontFamilyName"></param>
+        /// <param name="location"></param>
+        /// <param name="alignment"></param>
+        /// <param name="rotation"></param>
+        internal void InternalSetTextSettings(string value, bool bold, bool italic, double size, string fontFamilyName, DS.Point location, HorizontalTextAlignment alignment, double rotation)
+        {
+            this.Value = value;
+            this.Bold = bold;
+            this.Italic = italic;
+            this.FontSize = size;
+            this.FontFamilyName = fontFamilyName;
+            this.Location = location;
+            this.Alignment = alignment;
+            this.Rotation = rotation;
+        }
+
+        /// <summary>
+        /// Create the outline geometry based on the formatted text.
+        /// </summary>
+        /// <param name="text"></param>
+        /// <param name="bold"></param>
+        /// <param name="italic"></param>
+        /// <param name="family"></param>
+        /// <param name="size"></param>
+        /// <param name="location"></param>
+        /// <returns></returns>
+        private PathGeometry CreateText(string text, bool bold, bool italic, FontFamily family, double size, System.Windows.Point location)
+        {
+            System.Windows.FontStyle fontStyle = FontStyles.Normal;
+            FontWeight fontWeight = FontWeights.Medium;
+
+            if (bold == true) fontWeight = FontWeights.Bold;
+            if (italic == true) fontStyle = FontStyles.Italic;
+
+            FormattedText formattedText = new FormattedText(
+                text,
+                System.Globalization.CultureInfo.GetCultureInfo("en-us"),
+                FlowDirection.LeftToRight,
+                new Typeface(family, fontStyle,fontWeight,FontStretches.Normal),
+                size, System.Windows.Media.Brushes.Black 
+                );
+
+            Geometry textGeometry = formattedText.BuildGeometry(location);
+            return textGeometry.GetFlattenedPathGeometry();
+        }
+
+        /// <summary>
+        /// Text Tesselation
+        /// </summary>
+        /// <param name="package"></param>
+        /// <param name="parameters"></param>
+        [IsVisibleInDynamoLibrary(false)]
+        public new void Tessellate(IRenderPackage package, TessellationParameters parameters)
```

Comment:
hmm! because this functionality is quite useful it would be good to have it shared with DynamoCoreWPF, and cross platform.

I wonder if you could design this `TextElement` class to inherit from a `TextObject` class that is part of `DynamoCoreWpf` and of course the best outcome would be if somehow the `pathGeometry` from font/text was available in mono and in that case could be made to run cross platform and then this class could exist in `DynamoCore`... 

I think it's a worth file endeavor but we should focus on getting this stuff into Revit for now, so unless you have the interest and time it's not a top priority but thanks for thinking of it.